### ISSUE TITLE
Clock skew after upgrading from 3374.2.3 to 3510.2.1

### ISSUE THREAD

**Issue by gkoufoud** [flatcar/Flatcar#1031]
## Description
We have several EKS clusters and our EC2 instances use flatcar os. Since we upgraded from 3374.2.3 to 3510.2.1 there are prometheus alerts firing in all clusters about clock skew and particularly `abs(node_timex_offset_seconds) > 0.05`.

We noticed that systemd-timesyncd in the new version logs a lot of messages like:
```
ay 18 15:23:00 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:00 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Contacted time server 80.87.128.222:123 (0.flatcar.pool.ntp.org).
May 18 15:23:00 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:00 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:00 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:00 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:00 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Contacted time server 80.87.128.222:123 (0.flatcar.pool.ntp.org).
May 18 15:23:02 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:02 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Contacted time server 80.87.128.222:123 (0.flatcar.pool.ntp.org).
May 18 15:23:05 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:05 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:05 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:05 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Contacted time server 80.87.128.222:123 (0.flatcar.pool.ntp.org).
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Contacted time server 80.87.128.222:123 (0.flatcar.pool.ntp.org).
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Contacted time server 80.87.128.222:123 (0.flatcar.pool.ntp.org).
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Contacted time server 80.87.128.222:123 (0.flatcar.pool.ntp.org).
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Contacted time server 80.87.128.222:123 (0.flatcar.pool.ntp.org).
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
May 18 15:23:32 ip-10-206-56-161.eu-west-2.compute.internal systemd-timesyncd[1715]: Network configuration changed, trying to establish connection.
```

<img width="1749" alt="image" src="https://github.com/flatcar/Flatcar/assets/51472113/27981f3f-09af-46cf-970a-4bcb29e782ac">

## Impact

The impact is not yet clear but I suppose it will soon cause problems with Time Series Databases (TSDB) like prometheus and probably other workloads.

**Comment by andy-v-h** [comment]
I've debugged similar symptoms for a while now. 

My server had a unique set of resolved ntp pool servers from it's neighbors(i.e., what returned from `flatcar.pool.ntp.org`) , and it's skew, stratum and other details were wildly different than its rack-mates

I think some of the complexity in picking from `*.flatcar.pool.ntp.org` might be why we are drifting. If some nodes in a rack get to be strat 3 or 4, but another node is at strat 6 or 7, that's not ideal. So looking for consistent ntp sources became my next idea. I found time.apple.com to be a good source. and falling back to time.cloudflare.com.

**Comment by gkoufoud** [comment]
Thanks Andy...
The problem fixed after changing the ntp to `169.254.169.123` (AWS) and also set `PollIntervalMinSec=16` and `PollIntervalMaxSec=17`

**Comment by andy-v-h** [comment]
> Thanks Andy...
> The problem fixed after changing the ntp to `169.254.169.123` (AWS) and also set `PollIntervalMinSec=16` and `PollIntervalMaxSec=17`

Glad to hear it resolved your issue! 

Only additional comment is that Amazaon and Google respective cloud NTP servers aren't well suited for mixing with with NTP servers. Operators should avoid going that.
